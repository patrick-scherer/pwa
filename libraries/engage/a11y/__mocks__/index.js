/* eslint-disable require-jsdoc */
export const Section = ({ children }) => children;

/* eslint-enable require-jsdoc */
