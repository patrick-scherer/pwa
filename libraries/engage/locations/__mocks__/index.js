export const FulfillmentSelector = jest.fn().mockReturnValue('FulfillmentSelector');
export const FulfillmentSheet = jest.fn().mockReturnValue('FulfillmentSheet');
