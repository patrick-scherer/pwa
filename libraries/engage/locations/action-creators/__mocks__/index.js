/**
 * Take care that only necessary modules are imported for the test.
 * @shopgate/engage/constants exports the content of @shopgate/engage/constants/Portals.js which
 * imports from @shopgate/engage/product. This causes loading of a huge amount of files which are
 * not required here but massively increases run time of the tests.
 */
jest.mock('../../constants/', () => ({
  ...jest.requireActual('../../constants/ActionTypes'),
}));
