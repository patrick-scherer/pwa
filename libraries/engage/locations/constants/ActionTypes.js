// USER
export const SELECT_LOCATION = 'SELECT_LOCATION';
export const SELECT_GLOBAL_LOCATION = 'SELECT_GLOBAL_LOCATION';
export const STORE_FORM_INPUT = 'STORE_FORM_INPUT';
export const STORE_FULFILLMENT_METHOD = 'STORE_FULFILLMENT_METHOD';
export const SET_USER_SEARCH_COUNTRY_CODE = 'SET_USER_SEARCH_COUNTRY_CODE';
export const SET_USER_SEARCH_POSTAL_CODE = 'SET_USER_SEARCH_POSTAL_CODE';
export const SET_USER_SEARCH_GEOLOCATION = 'SET_USER_SEARCH_GEOLOCATION';

// LOCATIONS
export const REQUEST_LOCATIONS = 'REQUEST_LOCATIONS';
export const RECEIVE_LOCATIONS = 'RECEIVE_LOCATIONS';
export const ERROR_LOCATIONS = 'ERROR_LOCATIONS';

// PRODUCTION LOCATIONS
export const REQUEST_PRODUCT_LOCATIONS = 'REQUEST_PRODUCT_LOCATIONS';
export const RECEIVE_PRODUCT_LOCATIONS = 'RECEIVE_PRODUCT_LOCATIONS';
export const ERROR_PRODUCT_LOCATIONS = 'ERROR_PRODUCT_LOCATIONS';

// Does not belong here.
// TODO: Put into a 'fulfillment' domain in engage package.

// SUBMIT RESERVATION
export const SUBMIT_RESERVATION_REQUEST = 'SUBMIT_RESERVATION_REQUEST';
export const SUBMIT_RESERVATION_SUCCESS = 'SUBMIT_RESERVATION_SUCCESS';
export const SUBMIT_RESERVATION_ERROR = 'SUBMIT_RESERVATION_ERROR';
