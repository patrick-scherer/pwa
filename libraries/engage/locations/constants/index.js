// @flow
// ACTION TYPES
export {
  REQUEST_LOCATIONS,
  RECEIVE_LOCATIONS,
  ERROR_LOCATIONS,
  SELECT_LOCATION,
  STORE_FORM_INPUT,
  STORE_FULFILLMENT_METHOD,
  REQUEST_PRODUCT_LOCATIONS,
  RECEIVE_PRODUCT_LOCATIONS,
  ERROR_PRODUCT_LOCATIONS,
  SUBMIT_RESERVATION_REQUEST,
  SUBMIT_RESERVATION_SUCCESS,
  SUBMIT_RESERVATION_ERROR,
} from './ActionTypes';

// PIPELINES
export {
  SHOPGATE_STOREFRONT_GET_PRODUCT_LOCATIONS,
  SHOPGATE_STOREFRONT_CREATE_ORDER,
  SHOPGATE_STOREFRONT_GET_LOCATIONS,
  SHOPGATE_STOREFRONT_GET_CART,
  SHOPGATE_STOREFRONT_UPDATE_CART,
} from './Pipelines';

// PORTALS
export {
  PRODUCT_FULFILLMENT_SELECTOR,
  PRODUCT_LOCATION_STOCK_INFO,
} from './Portals';

export const DIRECT_SHIP = 'DIRECT_SHIP';
export const IN_STORE_PICKUP = 'inStorePickup';
export const ROPIS = 'ROPIS';
export const QUICK_RESERVE = 'quickReserve';
export const MULTI_LINE_RESERVE = 'multiLineReserve';

export {
  STAGE_SELECT_STORE,
  STAGE_RESERVE_FORM,
  STAGE_RESPONSE_SUCCESS,
  STAGE_RESPONSE_ERROR,
  STAGE_FULFILLMENT_METHOD,
} from './Stages';
