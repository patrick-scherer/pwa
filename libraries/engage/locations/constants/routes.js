export const STORE_FINDER_PATTERN = '/storefindernext';
