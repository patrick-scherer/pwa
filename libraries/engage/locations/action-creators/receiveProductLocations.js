import { RECEIVE_PRODUCT_LOCATIONS } from '../constants';

/**
 * Creates the dispatched RECEIVE_PRODUCT_LOCATIONS action object.
 * @param {string} productId The ID of the product.
 * @param {Array} locations List of product locations.
 * @param {Object} params Additional parameters.
 * @return {Object} The RECEIVE_PRODUCT_LOCATIONS action.
 */
const receiveProductLocations = (productId, locations, params) => ({
  type: RECEIVE_PRODUCT_LOCATIONS,
  productId,
  locations,
  params,
});

export default receiveProductLocations;
