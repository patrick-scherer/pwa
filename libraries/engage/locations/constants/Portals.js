// @flow
import { PRODUCT } from '@shopgate/pwa-common-commerce/product/constants/Portals';

// CONTENTS
const FULFILLMENT_SELECTOR = 'fulfillment-selector';
const LOCATION_STOCK_INFO = 'location-stock-info';

/* PRODUCT DETAILS */
export const PRODUCT_FULFILLMENT_SELECTOR = `${PRODUCT}.${FULFILLMENT_SELECTOR}`;
export const PRODUCT_LOCATION_STOCK_INFO = `${PRODUCT}.${LOCATION_STOCK_INFO}`;
