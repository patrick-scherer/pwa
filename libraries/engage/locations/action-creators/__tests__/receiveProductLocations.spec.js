import '../__mocks__';
import { RECEIVE_PRODUCT_LOCATIONS } from '../../constants';
import receiveProductLocations from '../receiveProductLocations';

describe('engage > locations > actions-creators', () => {
  test('receiveProductLocations', () => {
    const productCode = 'id123';
    const filters = {
      productCode: 'abc123',
    };
    const locations = [{ code: 'LOC1' }, { code: 'LOC2' }];
    const result = receiveProductLocations(productCode, filters, locations);
    expect(result).toEqual({
      type: RECEIVE_PRODUCT_LOCATIONS,
      productCode,
      locations,
      filters,
    });
    expect(result).toMatchSnapshot();
  });
});
