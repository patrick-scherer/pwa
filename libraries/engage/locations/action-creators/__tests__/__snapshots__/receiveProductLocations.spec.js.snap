// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`engage > locations > actions-creators receiveProductLocations 1`] = `
Object {
  "locations": Array [
    "a",
    "b",
    "c",
  ],
  "params": undefined,
  "productId": "id123",
  "type": "RECEIVE_PRODUCT_LOCATIONS",
}
`;
